app/src/main/java/io/pslab/activity/MultimeterActivity.java

```diff
@@ -393,26 +393,30 @@ public boolean onCreateOptionsMenu(Menu menu) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.record_pause_data:
-                if (ContextCompat.checkSelfPermission(this,
-                        Manifest.permission.WRITE_EXTERNAL_STORAGE)
-                        != PackageManager.PERMISSION_GRANTED) {
-                    ActivityCompat.requestPermissions(this,
-                            new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, MY_PERMISSIONS_REQUEST_STORAGE_FOR_DATA);
-                    return true;
-                }
-                if (recordData) {
-                    item.setIcon(R.drawable.ic_record_white);
-                    recordData = false;
-                    CustomSnackBar.showSnackBar(coordinatorLayout, getString(R.string.data_recording_paused), null, null, Snackbar.LENGTH_LONG);
-                } else {
-                    isDataRecorded = true;
-                    item.setIcon(R.drawable.pause_icon);
-                    if (!isRecordingStarted) {
-                        multimeterLogger = new CSVLogger(getString(R.string.multimeter));
-                        isRecordingStarted = true;
-                        recordData = true;
-                        CustomSnackBar.showSnackBar(coordinatorLayout, getString(R.string.data_recording_start), null, null, Snackbar.LENGTH_LONG);
+                if (scienceLab.isConnected()) {
+                    if (ContextCompat.checkSelfPermission(this,
+                            Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                            != PackageManager.PERMISSION_GRANTED) {
+                        ActivityCompat.requestPermissions(this,
+                                new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, MY_PERMISSIONS_REQUEST_STORAGE_FOR_DATA);
+                        return true;
                     }
+                    if (recordData) {
+                        item.setIcon(R.drawable.ic_record_white);
+                        recordData = false;
+                        CustomSnackBar.showSnackBar(coordinatorLayout, getString(R.string.data_recording_paused), null, null, Snackbar.LENGTH_LONG);
+                    } else {
+                        isDataRecorded = true;
+                        item.setIcon(R.drawable.pause_icon);
+                        if (!isRecordingStarted) {
+                            multimeterLogger = new CSVLogger(getString(R.string.multimeter));
+                            isRecordingStarted = true;
+                            recordData = true;
+                            CustomSnackBar.showSnackBar(coordinatorLayout, getString(R.string.data_recording_start), null, null, Snackbar.LENGTH_LONG);
+                        }
+                    }
+                } else {
+                    CustomSnackBar.showSnackBar(coordinatorLayout, getString(R.string.nothing_to_record), null, null, Snackbar.LENGTH_LONG);
```

Comment:
Since this else block is functioning only if PSLab is not connected, we can put the same "PSLab not connected" message here instead of a new one.